grakn-engine/src/main/java/ai/grakn/engine/controller/GraqlController.java

```diff
@@ -173,6 +175,9 @@ private String executeGraql(Request request, Response response) throws RetryExce
         //Define all anonymous variables in the query
         Optional<Boolean> defineAllVars = queryParameter(request, DEFINE_ALL_VARS).map(Boolean::parseBoolean);
 
+        //Used to check if serialisation of results is needed. When loading we skip this for the sake of speed
+        boolean skipSerialisation = parseBoolean(queryParameter(request, LOADING_DATA).orElse("false"));
```

Comment:
Maybe we could tie this to the whether "batch" is set as the mode? Or is that too magic.

Reply:
Too much magic I think. We mixing meanings up.